index.js

```diff
@@ -622,6 +634,15 @@ class Batch {
       node = await node.getChildNode(i)
     }
 
+    if (onlyIfChanged) {
```

Comment:
pretty sure this would never fire. the only time an existing key is found is above in c === 0

Reply:
I actually meant to add a question about this. In my testing I never triggered the loop, is there a branching factor in the bee tree that delays splits until a certain number of keys have been written? In my testing this was the branch that was exclusively hit.